src/commands/remind.ts

```diff
@@ -0,0 +1,124 @@
+import Roles from "../util/roles"
+import Reminder from "../entities/Reminder"
+import formatTimestamp from "../util/formatTimestamp"
+
+export default new Command({
+    name: "remind",
+    aliases: [],
+    description: "List and manage reminders.",
+    permission: [Roles.MANAGER],
+    usage: "",
+    subcommands: [
+        {
+            name: "add",
+            description: "Add a reminder.",
+            permission: [Roles.MANAGER],
+            usage: "<channel> <interval> <message>"
+        },
+        {
+            name: "edit",
+            description: "Edit a reminder.",
+            permission: [Roles.MANAGER],
+            usage: "<id> <message>"
+        },
+        {
+            name: "delete",
+            description: "Delete a reminder.",
+            permission: [Roles.MANAGER],
+            usage: "<id>"
+        }
+    ],
+    async run(this: Command, client: Client, message: Message, args: Args) {
+        const subcommand = args.consume().toLowerCase()
+
+        if (subcommand === "list" || !subcommand) {
+            const reminders = await Reminder.find()
+            const tidy: Record<string, { channel: string; message: string; end: Date }> =
+                {}
+
+            for (const reminder of reminders) {
+                if (!tidy[reminder.id])
+                    tidy[reminder.id] = { channel: "", message: "", end: new Date() }
+
+                tidy[reminder.id].channel = reminder.channel
+                tidy[reminder.id].message = reminder.message
+                tidy[reminder.id].end = reminder.end
+            }
+
+            let list = ""
+            for (const [id, { channel, message, end }] of Object.entries(tidy)) {
+                list += `${id}: \u200B \u200B <#${channel}> (next reminder ${formatTimestamp(end, "R")}) - ${message}\n`
+            }
+
+            return message.channel.sendSuccess({
+                author: { name: "Reminder list" },
+                description: list
+            })
+        }
+
+        if (subcommand === "add") {
+            const channel = await args.consumeChannel()
+            if (!channel) {
+                return message.channel.sendError("You must provide a channel!")
+            }
+
+            const time = args.consume().toLowerCase()
+            let millis
+            switch (time) {
+                case "test":
+                    millis = 1000 * 5
+                    break
+                case "weekly":
+                    millis = 1000 * 60 * 60 * 24 * 7 // 1 week (1000 milliseconds * 60 seconds * 60 minutes * 24 hours * 7 days)
+                    break
+                case "bi-weekly":
+                    millis = 1000 * 60 * 60 * 24 * 3.5 // 1 week (1000 milliseconds * 60 seconds * 60 minutes * 24 hours * 3.5 days)
+                    break
+                case "monthly":
+                    millis = 1000 * 60 * 60 * 24 * 30 // 30 days (1000 milliseconds * 60 seconds * 60 minutes * 24 hours * 30 days)
```

Comment:
```suggestion
                    millis = 1000 * 60 * 60 * 24 * 30 // 1 month (1000ms * 60s * 60m * 24h * 30d)
```

Reply:
this is fine for now, but maybe find a way to schedule this for the 1st of every month, instead of every 30 straight days? otherwise, it can land on the 31st, 2nd, etc. which would look kinda weird